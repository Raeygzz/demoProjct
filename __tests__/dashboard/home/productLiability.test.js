import React from 'react';
import 'react-native';

import ProductLiability from '../../../src/screens/dashboard/home/productLiability';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';


import Enzyme from 'enzyme';
import { shallow, mount, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


const navigation = {
  navigate: jest.fn()
}


describe('<ProductLiability />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(<ProductLiability />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('renders View, Text, StyleSheet, ScrollView, RupiSelect, NormalButton, InvertButton, RupiTextInput & clientIssueArr components using (enzyme)shallow,expect,find,exists (jest)toEqual,toBeDefined,toBeTruthy,', () => {
    const wrapper = shallow(<ProductLiability />);
    expect(wrapper.find('View').exists()).toEqual(true);
    expect(wrapper.find('Text').exists()).toEqual(true);
    expect(wrapper.find('StyleSheet')).toBeDefined();
    expect(wrapper.find('ScrollView')).toBeTruthy();
    expect(wrapper.find('RupiSelect').exists()).toEqual(true);
    expect(wrapper.find('InvertButton').exists()).toEqual(true);
    expect(wrapper.find('NormalButton').exists()).toEqual(true);
    expect(wrapper.find('RupiTextInput').exists()).toEqual(true);
    expect(wrapper.find('productDefectsArr').exists());
    expect(wrapper.find('productDefectsArr')).toBeTruthy();
  });


  it('test state and state value using (enzyme)shallow,state,expect & (jest)toEqual,toBeTruthy,toBeDefined,toBe', () => {
    const wrapper = shallow(<ProductLiability />);
    expect(wrapper.state()).toBeTruthy();
    expect(wrapper.state()).toBeDefined();
    expect(wrapper.state('productName')).toBe('');
    expect(wrapper.state().productDefect).toBe('');
    expect(wrapper.state('productDefectOptions')).toEqual([
      { id: 1, value: "Money Pending To Receive", label: "Money Pending To Receive" },
      { id: 2, value: "Money Pending To Pay", label: "Money Pending To Pay" },
      { id: 3, value: "Product Quantity To Demand", label: "Product Quantity In Demand" },
      { id: 4, value: "Product Defect", label: "Product Defect" },
      { id: 5, value: "Product Scheme", label: "Product Scheme" },
    ]);
    expect(wrapper.state('productNameOptions')).toEqual([
      { id: 1, value: 'ABC', label: 'ABC' },
      { id: 2, value: 'DEF', label: 'DEF' },
      { id: 3, value: 'GHI', label: 'GHI' },
      { id: 4, value: 'JKL', label: 'JKL' },
      { id: 5, value: 'MNO', label: 'MNO' }
    ]);
    expect(wrapper.instance().state.productDefectDescription).toBe('');
  })

  it('calls the productLiabilityHandler method & test onPress functionality along with tests this.props.navigation.navigate using (enzyme)shallow,instance,find,first,props,expect,exists, (React)forceUpdate, (jest)jest.spyOn,toHaveLength,toHaveBeenCalledWith', () => {
    const wrapper = shallow(<ProductLiability navigation={navigation} />);
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'productLiabilityHandler');

    instance.forceUpdate();
    wrapper.find('NormalButton').first().props().onItemPressed();

    expect(wrapper.exists());
    expect(spy.mock.calls).toHaveLength(1);
    expect(navigation.navigate).toHaveBeenCalledWith("Home");
  });


  it('calls the productLiabilityCancel method & test onPress functionality along with tests this.props.navigation.navigate using (enzyme)shallow,instance,find,first,props,expect,exists, (React)forceUpdate, (jest)jest.spyOn,toHaveLength,toHaveBeenCalledWith', () => {
    const wrapper = shallow(<ProductLiability navigation={navigation} />);
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'productLiabilityCancel');

    instance.forceUpdate();
    wrapper.find('InvertButton').first().props().onInvertItemPressed();

    expect(wrapper.exists());
    expect(spy.mock.calls).toHaveLength(1);
    expect(navigation.navigate).toHaveBeenCalledWith("Home");
  });

})
