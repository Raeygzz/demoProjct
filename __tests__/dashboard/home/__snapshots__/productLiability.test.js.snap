// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 20,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "white",
              "borderBottomWidth": 1,
              "height": 68,
            },
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "marginBottom": 20,
              "minHeight": 68,
              "overflow": "hidden",
              "paddingTop": 20,
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#71648E",
                  "fontSize": 12,
                },
                Object {
                  "position": "absolute",
                  "right": 18,
                  "top": 18,
                },
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Apercu_Regular",
                "fontSize": 14,
                "left": 0,
                "lineHeight": 17,
                "opacity": 0.68,
                "position": "absolute",
                "top": 16,
              }
            }
          >
            Product Name
          </Text>
          <Text
            minimumFontScale={16}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "bottom": 16,
                    "fontSize": 18,
                    "left": 0,
                    "position": "absolute",
                  },
                  Object {
                    "fontFamily": "Apercu-Medium",
                  },
                ],
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": 50,
                  "margin": 0,
                  "marginLeft": -9,
                  "opacity": 0,
                  "padding": 0,
                },
                Object {
                  "fontFamily": "Apercu-Bold",
                },
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "fontFamily": "Apercu Medium",
                  "fontSize": 18,
                  "padding": 0,
                },
                null,
                Object {
                  "color": "white",
                },
              ]
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Select Product Name",
                    "textColor": undefined,
                    "value": null,
                  },
                  Object {
                    "label": "ABC",
                    "textColor": undefined,
                    "value": "ABC",
                  },
                  Object {
                    "label": "DEF",
                    "textColor": undefined,
                    "value": "DEF",
                  },
                  Object {
                    "label": "GHI",
                    "textColor": undefined,
                    "value": "GHI",
                  },
                  Object {
                    "label": "JKL",
                    "textColor": undefined,
                    "value": "JKL",
                  },
                  Object {
                    "label": "MNO",
                    "textColor": undefined,
                    "value": "MNO",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "white",
              "borderBottomWidth": 1,
              "height": 68,
            },
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "marginBottom": 20,
              "minHeight": 68,
              "overflow": "hidden",
              "paddingTop": 20,
              "position": "relative",
            },
            undefined,
          ]
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#71648E",
                  "fontSize": 12,
                },
                Object {
                  "position": "absolute",
                  "right": 18,
                  "top": 18,
                },
                Object {
                  "fontFamily": "simple-line-icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Apercu_Regular",
                "fontSize": 14,
                "left": 0,
                "lineHeight": 17,
                "opacity": 0.68,
                "position": "absolute",
                "top": 16,
              }
            }
          >
            Product Defect
          </Text>
          <Text
            minimumFontScale={16}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 16,
                },
                Array [
                  Object {
                    "bottom": 16,
                    "fontSize": 18,
                    "left": 0,
                    "position": "absolute",
                  },
                  Object {
                    "fontFamily": "Apercu-Medium",
                  },
                ],
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "height": 50,
                  "margin": 0,
                  "marginLeft": -9,
                  "opacity": 0,
                  "padding": 0,
                },
                Object {
                  "fontFamily": "Apercu-Bold",
                },
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "fontFamily": "Apercu Medium",
                  "fontSize": 18,
                  "padding": 0,
                },
                null,
                Object {
                  "color": "white",
                },
              ]
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Select Product Defect",
                    "textColor": undefined,
                    "value": null,
                  },
                  Object {
                    "label": "Money Pending To Receive",
                    "textColor": undefined,
                    "value": "Money Pending To Receive",
                  },
                  Object {
                    "label": "Money Pending To Pay",
                    "textColor": undefined,
                    "value": "Money Pending To Pay",
                  },
                  Object {
                    "label": "Product Quantity In Demand",
                    "textColor": undefined,
                    "value": "Product Quantity To Demand",
                  },
                  Object {
                    "label": "Product Defect",
                    "textColor": undefined,
                    "value": "Product Defect",
                  },
                  Object {
                    "label": "Product Scheme",
                    "textColor": undefined,
                    "value": "Product Scheme",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "marginBottom": 20,
              "minHeight": 68,
              "overflow": "hidden",
              "paddingTop": 20,
              "position": "relative",
            },
            undefined,
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Apercu_Regular",
              "fontSize": 14,
              "left": 0,
              "lineHeight": 17,
              "opacity": 0.68,
              "position": "absolute",
              "top": 26,
            }
          }
        >
          Product Description
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          multiline={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={null}
          placeholderTextColor="rgba(255,255,255,0.7)"
          rejectResponderTermination={true}
          returnKeyType="done"
          selectionColor="#71648E"
          style={
            Array [
              Object {
                "borderBottomColor": "white",
                "borderBottomWidth": 1,
                "color": "white",
                "fontFamily": "Apercu Medium",
                "fontSize": 18,
                "padding": 0,
              },
              null,
              undefined,
              Object {
                "borderBottomColor": "white",
                "height": 37,
                "lineHeight": 12,
              },
              null,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 30,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffbcbc",
                "borderColor": "red",
                "borderWidth": 1,
                "height": 100,
                "justifyContent": "center",
                "margin": 10,
                "width": 300,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "margin": 20,
              }
            }
          >
            TouchableNativeFeedback is not supported on this platform!
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffbcbc",
                "borderColor": "red",
                "borderWidth": 1,
                "height": 100,
                "justifyContent": "center",
                "margin": 10,
                "width": 300,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "margin": 20,
              }
            }
          >
            TouchableNativeFeedback is not supported on this platform!
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "red",
            "marginTop": 30,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Cleansing Milk
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Money Pending To Pay :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "green",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Ayur Cream
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Money Pending To Receive :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "lightgreen",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Acnamouist Cream
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Product Quantity To Demand :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "pink",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Vaseline Cream 
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Product Defect :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "darkgrey",
            "marginVertical": 20,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "marginBottom": 20,
              "textAlign": "center",
            }
          }
        >
          Sanitary Pad
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
            }
          }
        >
          Product Quantity To Demand :
        </Text>
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
               Rs. 20000 is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor. Is to be paid to the vendor.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
